omnigibson/scenes/scene_base.py

```diff
@@ -169,85 +205,138 @@ def object_registry_group_keys(self):
+                obj.prebuild(stage)
+
+            stage.Save()
+            del stage
+
+            PREBUILT_USDS[self.scene_file] = usd_path
+
+        # Copy the prebuilt USD to a new path
+        decrypted_fd, instance_usd_path = tempfile.mkstemp(os.path.basename(self.scene_file) + ".usd", dir=og.tempdir)
+        os.close(decrypted_fd)
+        shutil.copyfile(PREBUILT_USDS[self.scene_file], instance_usd_path)
+        return instance_usd_path
+
     def _load(self):
         """
         Load the scene into simulator
         The elements to load may include: floor, building, objects, etc.
         """
-        # Create collision group for fixed base objects' non root links, root links, and building structures
-        CollisionAPI.create_collision_group(col_group="fixed_base_nonroot_links", filter_self_collisions=False)
-        # Disable collision between root links of fixed base objects
-        CollisionAPI.create_collision_group(col_group="fixed_base_root_links", filter_self_collisions=True)
-        # Disable collision between building structures
-        CollisionAPI.create_collision_group(col_group="structures", filter_self_collisions=True)
+        # There's nothing to load for the base scene. Subclasses can implement this method.
+        # We simply check that the simulator has a floor plane.
+        assert og.sim.floor_plane, "Simulator must have a floor plane if using an empty Scene!"
 
-        # Disable collision between building structures and 1. fixed base objects, 2. attached objects
-        CollisionAPI.add_group_filter(col_group="structures", filter_group="fixed_base_nonroot_links")
-        CollisionAPI.add_group_filter(col_group="structures", filter_group="fixed_base_root_links")
+    def _load_systems(self):
+        system_dir = os.path.join(gm.DATASET_PATH, "systems")
 
-        # We just add a ground plane if requested
-        if self._use_floor_plane:
-            self.add_ground_plane(color=self._floor_plane_color, visible=self._floor_plane_visible)
+        if os.path.exists(system_dir):
+            system_names = os.listdir(system_dir)
+            for system_name in system_names:
+                self.system_registry.add(create_system_from_metadata(system_name=system_name))
 
-        # Also add skybox if requested
-        if self._use_skybox:
-            self._skybox = LightObject(
-                prim_path="/World/skybox",
-                name="skybox",
-                category="background",
-                light_type="Dome",
-                intensity=1500,
-                fixed_base=True,
-            )
-            og.sim.import_object(self._skybox, register=False)
-            self._skybox.color = (1.07, 0.85, 0.61)
-            # Default texture to use for skybox
-            self._skybox.texture_file_path = f"{gm.ASSET_PATH}/models/background/sky.jpg"
+        from omnigibson import systems
+
+        cloth_system = systems.Cloth(name="cloth")
```

Comment:
@ChengshuLi do you know if we'll have any overhead for having N cloth systems (or in general, M systems * N scenes)?

Reply:
Hopefully it should strictly make things faster by avoiding collision checks between things in diff scenes but maybe that's a bad assumption.

Reply:
TODO: look into this